tests/src/providers/testqgspostgresprovider.cpp

```diff
@@ -190,5 +194,108 @@ void TestQgsPostgresProvider::testDecodeDateTimes()
 
 }
 
+void TestQgsPostgresProvider::testQuotedValueBigInt()
+{
+  QgsFields fields;
+  QList<int> pkAttrs;
+  QVariantList vlst;
+
+  std::shared_ptr< QgsPostgresSharedData > sdata( new QgsPostgresSharedData() );
+
+  QgsField f0, f1, f2, f3;
+
+  // 4 byte integer
+  f0.setName( "fld_integer" );
+  f0.setType( QVariant::Int );
+  f0.setTypeName( "int" );
+
+  fields.append( f0 );
+  pkAttrs.append( 0 );
+  vlst.append( 42 );
+
+  // for positive integers, fid  == the value, there is no map.
+  sdata->insertFid( 42, vlst );
+
+  QCOMPARE( QgsPostgresUtils::whereClause( 42, fields, NULL, QgsPostgresPrimaryKeyType::PktInt, pkAttrs, std::shared_ptr<QgsPostgresSharedData>( sdata ) ), QString( "\"fld_integer\"=42" ) );
+
+  // 8 byte integer
+  f1.setName( "fld_bigint" );
+  f1.setType( QVariant::LongLong );
+  f1.setTypeName( "int8" );
+
+  fields.clear();
+  pkAttrs.clear();
+  vlst.clear();
+
+  fields.append( f1 );
+  pkAttrs.append( 0 );
+  vlst.append( -9223372036854775800LL ); // way outside int4 range
+
+  sdata->clear();
+  sdata->insertFid( 1LL, vlst );
+
+  QCOMPARE( QgsPostgresUtils::whereClause( 1LL, fields, NULL, QgsPostgresPrimaryKeyType::PktInt64, pkAttrs, std::shared_ptr<QgsPostgresSharedData>( sdata ) ), QString( "\"fld_bigint\"=-9223372036854775800" ) );
+
+  // double
+  f2.setName( "fld_double" );
+  f2.setType( QVariant::Double );
+  f2.setTypeName( "float8" );
+
+  fields.clear();
+  pkAttrs.clear();
+  vlst.clear();
+
+  fields.append( f2 );
+  pkAttrs.append( 0 );
+  vlst.append( 3.141592741 );
```

Comment:
I like this one :+1:

Reply:
A little homage to the most well rounded release of QGIS yet 😁